src/Compilers/CSharp/Portable/BoundTree/TupleBinaryOperatorInfo.cs

```diff
@@ -111,5 +111,37 @@ internal override TreeDumperNode DumpCore()
             }
 #endif
         }
+
+        /// <summary>
+        /// Represents an element-wise null/null comparison.
+        /// For instance, `(null, ...) == (null, ...)`.
+        /// </summary>
+        internal class NullNull : TupleBinaryOperatorInfo
+        {
+            internal readonly BinaryOperatorKind Kind;
+
+            internal NullNull(BinaryOperatorKind kind, TypeSymbol leftConvertedTypeOpt, TypeSymbol rightConvertedTypeOpt)
+                : base(leftConvertedTypeOpt, rightConvertedTypeOpt)
+            {
+                Kind = kind;
+            }
+
+            internal override KindEnum InfoKind
+                => KindEnum.NullNull;
+
+#if DEBUG
+            internal override TreeDumperNode DumpCore()
+            {
+                return new TreeDumperNode("nullnull", value: Kind, children: null);
+            }
+#endif
+        }
+
+        internal enum KindEnum
```

Comment:
>KindEnum [](start = 22, length = 8)

TupleBinaryOperatorInfoKind? Also, consider moving this declaration to the beginning of the file